lib/boxen/preflight/creds.rb

```diff
@@ -68,12 +70,27 @@ def run
     fetch_login_and_password
     tokens = get_tokens
 
-    unless auth = tokens.detect { |a| a.note == "Boxen" }
-      auth = tmp_api.create_authorization \
-        :note => "Boxen",
-        :scopes => %w(repo user),
-        :headers => headers
-    end
+    # Boxen now supports the updated GitHub Authorizations API by using a unique
+    # `fingerprint` for each Boxen installation for a user. We delete any older
+    # authorization that does not make use of `fingerprint` so that the "legacy"
+    # authorization doesn't persist in the user's list of personal access
+    # tokens.
+    legacy_auth = tokens.detect { |a| a.note == "Boxen" && a.fingerprint == nil }
```

Comment:
This will take care of the base case for most users. But, there are edge cases we can't solve here. For example:

* User updates to this new version of boxen on Mac1 and logs in. 
* Legacy token they already had is deleted and a new token is created specifically for Mac1.
* Mac2 logs in using an older version of Boxen, and ends up creating yet a new "legacy token" that has no fingerprint.
* Mac3 logs in using an older version of Boxen and can't get the plaintext legacy token that Mac2 ended up creating. 
* Mac3 is stuck unless/until they either delete the other legacy token or upgrade their version of Boxen.

So, while we can't solve all problems with this change, it will be ideal for most users and other users will be no worse off than if we didn't try to clear the legacy token for them.